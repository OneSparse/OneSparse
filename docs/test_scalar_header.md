# Scalar

This documentation is also tests for the code, the examples below
show the literal output of these statements from Postgres.

Some setup to make sure warnings are shown, and that the extension
is installed.
``` postgres-console
set client_min_messages = 'WARNING';
create extension if not exists onesparse;
```
Describe the scalar type
``` postgres-console
\dT+ scalar
                                        List of data types
  Schema   |  Name  | Internal name | Size | Elements |  Owner   | Access privileges | Description 
-----------+--------+---------------+------+----------+----------+-------------------+-------------
 onesparse | scalar | scalar        | var  |          | postgres |                   | 
(1 row)

```
print a scalar
``` postgres-console
select print('i4:42'::scalar);
                    print                    
---------------------------------------------
                                            +
   1x1 GraphBLAS int32_t scalar, full by col+
   A->scalar, 1 entry, memory: 240 bytes    +
   iso value:   42                          +
                                            +
 
(1 row)

```
print a scalar
``` postgres-console
select print('i4:42'::scalar, 5);
                    print                    
---------------------------------------------
                                            +
   1x1 GraphBLAS int32_t scalar, full by col+
   A->scalar, 1 entry, memory: 240 bytes    +
   iso value:   42                          +
                                            +
     (0,0)   42                             +
 
(1 row)

```
Duplicate a scalar
``` postgres-console
select dup('i4:42'::scalar);
  dup  
-------
 i4:42
(1 row)

```
Wait for a scalar to complete in non-blocking mode
``` postgres-console
select wait('i4:42'::scalar);
 wait  
-------
 i4:42
(1 row)

```
Clear a scalar, deleting its stored element.
``` postgres-console
select clear('i4:42'::scalar);
 clear 
-------
 i4:
(1 row)

```