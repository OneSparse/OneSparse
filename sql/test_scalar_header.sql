set client_min_messages = 'WARNING';
create extension if not exists onesparse;
set search_path = 'onesparse';

select 'i2:1'::scalar;
select 'i4:2'::scalar;
select 'i8:3'::scalar;
select 'f4:2.0'::scalar;
select 'f8:3.0'::scalar;
