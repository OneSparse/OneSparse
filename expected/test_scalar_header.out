set client_min_messages = 'WARNING';
create extension if not exists onesparse;
\dT+ scalar
                                        List of data types
  Schema   |  Name  | Internal name | Size | Elements |  Owner   | Access privileges | Description 
-----------+--------+---------------+------+----------+----------+-------------------+-------------
 onesparse | scalar | scalar        | var  |          | postgres |                   | 
(1 row)

select dup('i4:42'::scalar);
  dup  
-------
 i4:42
(1 row)

select wait('i4:42'::scalar);
 wait  
-------
 i4:42
(1 row)

select clear('i4:42'::scalar);
 clear 
-------
 i4:
(1 row)

